src/nb_type.cpp

```diff
@@ -847,6 +988,7 @@ static PyTypeObject *nb_type_tp(size_t supplement) noexcept {
 
         PyType_Slot slots[] = {
             { Py_tp_base, &PyType_Type },
+            { Py_tp_finalize, (void *) nb_type_finalize },
```

Comment:
I was curious about the split into finalization and deallocation. Why was this needed?